lte/gateway/c/core/oai/tasks/s1ap/s1ap_state_converter.cpp

```diff
@@ -60,6 +60,14 @@ void S1apStateConverter::state_to_proto(s1ap_state_t* state, S1apState* proto) {
     FREE_HASHTABLE_KEY_ARRAY(keys);
   }
 
+  keys                        = hashtable_ts_get_keys(&state->enbs);
+  uint32_t expected_enb_count = keys ? keys->num_keys : 0;
+  if (expected_enb_count != state->num_enbs) {
+    OAILOG_ERROR(
+        LOG_S1AP, "Updating num_eNBs from maintained to actual count %u->%u",
+        state->num_enbs, expected_enb_count);
+    state->num_enbs = expected_enb_count;
```

Comment:
We could also remove this state from the persistence layer and then just reconstruct it, but this helps us make sure that our stats counting is sane as well.